src/braket/circuits/circuit.py

```diff
@@ -472,7 +473,9 @@ def add_instruction(
 
         if self._check_for_params(instruction):
             for param in instruction.operator.parameters:
-                if isinstance(param, FreeParameterExpression):
+                if isinstance(param, FreeParameterExpression) and isinstance(
+                    param.expression, Expr
+                ):
```

Comment:
Wouldn't it be better to ensure that `isinstance(expression, Expr)` on `FreeParameterExpression` instantiation? The you don't have to do this check for both `Circuit` and `PulseSequence`.